modules/thermo-cycler/production/firmware_uploader.py

```diff
@@ -56,11 +56,13 @@ def find_bootloader_drive():
     # takes around 4 seconds for TCBOOT to register in /Volumes
     retries = 7
     while retries:
-        for d in uf2conv.get_drives():
-            if "TCBOOT" in d:
-                print("Bootloader Volume found")
-                return d
         print("Searching bootloader...")
+        for d in uf2conv.get_drives():
+            with open(os.path.join(d + '/INFO_UF2.TXT'), mode='r') as f:
```

Comment:
You're probably going to get a FileNotFound error here if the file doesn't exist in that root. Also, `os.path.join()` takes multiple arguments (like `os.path.join(d, 'INFO_UF2.TXT')`. You may want to consider using pathlib here.

Reply:
Also maybe move this into `uf2conv`? it's used both here and in the serial and firmware uploader, so might as well centralize it